File: app/src/main/java/com/anthonymandra/rawdroid/data/SubjectEntity.java
package com.anthonymandra.rawdroid.data;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;

@Entity(tableName = "xmp_subject",
		indices = @Index(value = "id"))
public class SubjectEntity extends PathEntity
{
	SubjectEntity() {}
	SubjectEntity(String path, int depth, long parent, String name, Long recent)
	{
		super(path, depth, parent);
		this.name = name;
		this.recent = recent;
	}

	SubjectEntity(Long id, String path, int depth, long parent, String name, Long recent)
	{
		super(id, path, depth, parent);
		this.name = name;
		this.recent = recent;
	}

	public String name;
	public Long recent;
}

File: app/src/main/java/com/anthonymandra/rawdroid/data/SubjectEntity.kt
package com.anthonymandra.rawdroid.data

import androidx.room.Entity
import androidx.room.Index
import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
@Entity(
    tableName = "xmp_subject",
    indices = [ Index(value = [ "id" ]) ])
data class SubjectEntity(
    var name: String = "",
    var recent: Long = 0
): PathEntity(), Parcelable
